src/corelibs/U2Designer/src/dashboard/DashboardWidget.cpp

```diff
@@ -280,19 +281,52 @@ void DashboardFileButton::addUrlActionsToMenu(QMenu *menu, const QString &url, b
     menu->addAction(openFileAction);
 }
 
+/**
+ * Finds a file in the given dashboard dir by path suffix of the 'fileInfo'.
+ * Returns new file info or the old one if the file detection algorithm is failed.
+ * This method is designed to find dashboard output files in moved dashboard.
+ */
+static QFileInfo findFileOpenCandidateInTheDashboardOutputDir(const QFileInfo &dashboardDirInfo, const QFileInfo &fileInfo) {
```

Comment:
Please, separate declaration to the ".h" file.

Reply:
This function is not not designed to be a public API or to be a class member (depend on a class context). 
It is designed to be a stateless local function. I prefer limiting the scope/context where possible.

If you have a strong opinion that the function should be declared as a public one in the .h file I will move it.